feat: issue 212 added support for existing serviceID

vbontempi opened the pull request:
### Description

Added support for an existing ServiceID to be used when creating a COS instance and the related resource key

### Types of changes in this PR

- support for an existing ServiceID into COS resource key
- added existing serviceID into existing resources test

#### No release required

- [ ] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other changes that don't affect Terraform code

#### Release required

- [ ] Bug fix (patch release (`x.x.X`): Change that fixes an issue and is compatible with earlier versions)
- [X] New feature (minor release (`x.X.x`): Change that adds functionality and is compatible with earlier versions)
- [ ] Breaking change (major release (`X.x.x`): Change that is likely incompatible with previous versions)

##### Release notes content

This change allows to create the COS credentials resource key binding it with a pre-existing IAM ServiceID

---

### Checklist for reviewers

- [X] The PR references a GitHub issue https://github.com/terraform-ibm-modules/terraform-ibm-cos/issues/212
- [X] If relevant, a test for the change is included or updated with this PR.
- [X] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Merge by using "Squash and merge".
- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author.

    The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).

ocofaigh reviewed (changes requested):
Could we update the `existing-resources` example to create a new service ID and then pass it in to the module using the new `resource_key_existing_serviceid_name` variable? That way we will al least test it in our tests

ocofaigh reviewed (changes requested):
Can we update the input variable to take in the Service ID CRN instead of name please. Name is not a computed value, and so we should avoid using non computed values like this as terraform may try to use it before it exists.
That way consumers won't have to use explicit `depends_on` which is not recommended by terraform

vbontempi commented:
> Can we update the input variable to take in the Service ID CRN instead of name please. Name is not a computed value, and so we should avoid using non computed values like this as terraform may try to use it before it exists. That way consumers won't have to use explicit `depends_on` which is not recommended by terraform

Modified code to accept serviceID crn in input instead of the name
Modified test accordingly

ocofaigh commented on variables.tf:
```diff
@@ -29,6 +29,12 @@ variable "create_hmac_key" {
 }
 
+variable "resource_key_existing_serviceid_crn" {
+  description = "CRN of existing serviceID to bind with resource key to be created"
```
what happens if this is left at null? A new service Id is created? If so, lets say that in the description here please

vbontempi replied:
yes, if null the behavior is exactly the same of current one where no serviceID is specified

ocofaigh commented on examples/existing-resources/main.tf:
```diff
@@ -30,19 +30,32 @@ module "key_protect_all_inclusive" {
-  activity_tracker_crn               = null
+  depends_on = [
+    ibm_iam_service_id.resource_key_existing_serviceid
+  ]
```
remove the depends_on

vbontempi commented:
> final comments

addressed

terraform-ibm-modules-ops commented:
:tada: This PR is included in version 5.9.0 :tada:

The release is available on [GitHub release](https://github.com/terraform-ibm-modules/terraform-ibm-cos/releases/tag/v5.9.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
